x-pack/plugins/index_lifecycle_management/public/application/services/policies/policy_serialization.test.ts

```diff
@@ -367,4 +368,135 @@ describe('Policy serialization', () => {
     serializePolicy(deserializedPolicy, originalPolicy);
     expect(originalPolicy).toEqual(originalClone);
   });
+
+  test('serialize a policy using "best_compression" codec for forcemerge', () => {
```

Comment:
I think it is worth adding a test that asserts our serialisation will remove the "best_compression" setting from a policy that originally has it specified. This will ensure that we do not have a dangling "best_compression" setting.